#161340194 Reporting an article

Musacoli opened the pull request:
#### What does this PR do?
Users should be able to report an article

#### Description of Task to be completed?
As an authenticated user I should be able to report articles that violate terms of the agreement or are plagiarised to the authors haven admin for removal and or any other actions as per the terms of engagement

#### How should this be manually tested?
After cloning the repo to your machine, cd into it and run this snippet in the terminal
`python manage.py runserver`

Open your Postman or Postico and feed it with the following URL after having logged in and been authenticated.
`http://127.0.0.1:8000/api/articles/<slug>/report/`

The BODY of your POST request should contain the following content
`{ "report":
                    { "reason": "the reason as to why the article is being reported"} 
                }`

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
#161340194

#### Screenshots (if appropriate)
#### Questions:

Commit:
[Feature #161340178] Create articles app

Commit:
[Chore #161340170] Install django-nose

Commit:
[Chore #161340170] Enable coverage percentage to show

Commit:
[Feature #161340178] Create articles model

Commit:
[Feature #161340178] Delete tests file

Commit:
[Feature #161340178] Add create article view

Commit:
[Feature #161340178] Add url for article creation

Commit:
[Feature #161340178] Add articles app to installed apps

Commit:
[Feature #161340178] Add  serializsers for articles app

Commit:
[Feature #161340178] Declare the urls for the articles app

Commit:
[Feature #161340178] Declare urls for the articles app

Commit:
[Feature #161340178] Install django-taggit

Commit:
[Feature #161340178] Description should not be a primary key

Commit:
[Feature #161340178] Remove bio and image for the user

Commit:
[Feature #161340178] Remove nose as the test_runner

Commit:
[Feature #161340178] Add initial migrations

Commit:
[Feature #161340178] Add initial migrations

Commit:
[Feature #161340178] Edit the rendererer to return a list of articles

Commit:
[Feature #161340178] Add update article serializer

Commit:
[Feature #161340178] Create update article view

Commit:
[Feature #161340178] Include a back slash on the articles url

Commit:
[Feature #161340178] Add path to update article view

Musacoli committed:
feat(api) report an article

Musacoli committed:
Update to test settings

Musacoli committed:
Update authors/apps/reporter/views.py

Musacoli committed:
Update authors/apps/reporter/tests.py

Musacoli committed:
Update .travis.yml

Musacoli committed:
Update requirements.txt

Musacoli committed:
Merge branch 'develop' into ft-reporting-an-article-161340194

SoniaRMK commented:
@Musacoli Kindly resolve the merge conflicts, rename your PR according to the PR naming conventions per the [Andela Engineering book](https://github.com/andela/engineering-playbook/tree/master/5.%20Developing/Conventions) and make sure your coverage doesn't drop below what is on Develop.

Also, your commits should be squashed to 1 commit. And the commit message should follow the conventions in the above mentioned book.

Musacoli committed:
Merge branch 'develop' into ft-reporting-an-article-161340194

Musacoli committed:
Update .travis.yml

Musacoli committed:
Update .travis.yml

Musacoli committed:
Update authors/test_settings.py

Musacoli committed:
test (reporter): added and fixed coverage

- ensure that reporter has running test
- added notification system for when user reports an article

[Delivers #130635935]

Musacoli committed:
Merge remote-tracking branch 'origin/ft-create-articles-161340178' into ft-reporting-an-article-161340194

Musacoli committed:
"feat(api) report an article"

Musacoli committed:
Merge branch 'ft-reporting-an-article-161340194' of https://github.com/andela/ah-backend-odin into ft-reporting-an-article-161340194

Musacoli committed:
fix squash commit

Musacoli committed:
Merge branch 'ft-reporting-an-article-161340194' of https://github.com/andela/ah-backend-odin into ft-reporting-an-article-161340194

Musacoli committed:
feat (api) User sjould be able to report an article

Musacoli committed:
Merge branch 'ft-reporting-an-article-161340194' of https://github.com/andela/ah-backend-odin into ft-reporting-an-article-161340194

Musacoli committed:
Merge branch 'ft-reporting-an-article-161340194' of https://github.com/andela/ah-backend-odin into ft-reporting-an-article-161340194

krmroland commented on .travis.yml:
```diff
@@ -12,6 +12,7 @@ before_script:
   - pip install -r requirements.txt
   - pip install coveralls
   - psql -c 'create database authors;' -U postgres
+  - python manage.py makemigrations  reporter --noinput
```
Duplicate line here

Musacoli replied:
this is necessary as the migrations for reporter weren't running on Travis without that line

Musacoli committed:
Merge branch 'ft-reporting-an-article-161340194' of https://github.com/andela/ah-backend-odin into ft-report-an-article-161340194
